src/ui/inventory.js

```diff
@@ -0,0 +1,182 @@
+                    pointer.worldY > y + menuBox.height / 2 ||
+                    pointer.worldX < x - menuBox.width / 2 ||
+                    pointer.worldY < y - menuBox.height / 2
+                ) {
+                    this.menu.destroy();
+                }
+            });
+
+        // Add text options
+        // Have to use two separate texts per option for different clors
+        let options = [menuBox];
+        let optionsY = y -45;
+
+        // Generate dynamic list of actions (use, wield, bury, etc.)
+        item.actions.forEach(action => {
+            optionsY += 15;
+            let itemText = this.scene.add.text(x + 10, optionsY, item.name, FONTS.ITEM_NAME);
+            let actionText = this.scene.add.text(x - 48, optionsY, action, FONTS.OPTIONS_MENU)
+                .setInteractive()
+                .setDepth(5)
+                .on("pointerdown", () => {
+                    item[action]();
+                    this.menu.destroy();
+                });
+
+            options.push(actionText);
+            options.push(itemText);
+        });
+
+        // All items have drop, examine, cancel
+        let dropItem = this.scene.add.text(x + 10, optionsY + 15, item.name, FONTS.ITEM_NAME);
+        let dropText = this.scene.add.text(x - 48, optionsY + 15, "Drop", FONTS.OPTIONS_MENU)
+            .setInteractive()
+            .on("pointerdown", () => {
+                    item.drop();
+                    this.menu.destroy();
+                });
+        options.push(dropItem);
+        options.push(dropText);
+            
+        let examineItem = this.scene.add.text(x + 10, optionsY + 30, item.name, FONTS.ITEM_NAME);
+        let examineText = this.scene.add.text(x - 48, optionsY + 30, "Examine", FONTS.OPTIONS_MENU)
+            .setInteractive()
+            .on("pointerdown", () => {
+                item.examine();
+                this.menu.destroy();
+            });
+        options.push(examineItem);
+        options.push(examineText);
+
+        let cancelText = this.scene.add
+            .text(x - 48, optionsY + 45, "Cancel", FONTS.OPTIONS_MENU)
+            .setInteractive()
+            .on("pointerdown", () => {
+                console.log("cancel");
+                this.menu.destroy();
+            });
+        options.push(cancelText);
+
+        // Group objects together
+        this.menu = this.scene.add
+            .container(0, 0, options)
+            .setDepth(5);
+    }
+
+    highlightItem(index) {
+        if (index == this.curSelectedItemIndex) {
+            this.curSelectedItemIndex = -1;
+        }
+        else {
+            // Use this ugly method to see if obj is empty
+            if (this.curSelectedItemIndex >= 0 && Object.keys(this.inventory[this.curSelectedItemIndex]).length) {
+                this.inventory[this.curSelectedItemIndex].highlightItem();
+            }
+            this.curSelectedItemIndex = index;
+        }
+    }
+
```

Comment:
Eventually we'll want dragging/swapping functionality

Reply:
Yeah that's the purpose of the methods in ItemType for changing their index and stuff. I've looked into how I'd do it and phaser has built in things called "drop zones" that allow you to drag and drop and it locks into a spot. 

When we do that we would call those methods plus update this variable.